airflow/contrib/hooks/gcp_transfer_hook.py

```diff
@@ -56,52 +85,282 @@ def get_conn(self):
                                http=http_authorized, cache_discovery=False)
         return self._conn
 
-    def create_transfer_job(self, description, schedule, transfer_spec, project_id=None):
-        transfer_job = {
-            'status': 'ENABLED',
-            'projectId': project_id or self.project_id,
-            'description': description,
-            'transferSpec': transfer_spec,
-            'schedule': schedule or self._schedule_once_now(),
-        }
-        return self.get_conn().transferJobs().create(body=transfer_job).execute()
-
-    def wait_for_transfer_job(self, job):
+    @GoogleCloudBaseHook.convert_http_exception
+    def create_transfer_job(self, body):
+        """
+        Creates a transfer job that runs periodically.
+
+        :param body: A request body, as described in
+            https://cloud.google.com/storage-transfer/docs/reference/rest/v1/transferJobs/patch#request-body
+        :type body: dict
+        :return: If successful, TransferJob.
+        :rtype: dict
+        """
+        body = self._append_project_id(body, 'body project_id parameter')
```

Comment:
In all 4 usages of this method the `project_id parameter` part of the "alternative" parameter is repeated.
Maybe we could extract these words into the method body and simply method invocations?